### DISCUSSION TITLE
Distribute Lottie as XCFramework on Github releases

### DISCUSSION THREAD

**Discussion by anuragajwani** [airbnb/lottie-ios#1743] [General]
Hi,

We are already consuming Lottie through Cocoapods. We are looking to reduce the compilation times of our app by consuming compiled XCFrameworks of third party dependencies. Some dependencies offers these in their release tabs (i.e. [RxSwift](https://github.com/ReactiveX/RxSwift/releases/tag/6.5.0)).

I would like to request/help building and distributing built xcframework when new releases are published.

Thanks,
Anurag

**Comment by calda** [comment]
Lottie already supports four package managers that have to be updated on each release (Cocoapods, Carthage, SPM, NPM) so I'm somewhat reluctant to add a fifth release method.

Have you considered using [Carthage](https://github.com/Carthage/Carthage) instead of Cocoapods? It builds `.xcframework`s instead of adding the source of your dependencies to your project directory. Alternatively you could export a `.xcframework` pretty easily by opening the `Lottie.xcworkspace` in this repo and exporting one.

**Comment by anuragajwani** [comment]
My only concern with using Carthage is that other dependencies are dropping support for it. I would prefer not to be using it given the risk. I do understand your concerns. Are you planning to drop Carthage support in the future?

**Reply by calda** [reply]
We don't currently have any plans to drop support for Carthage -- there is still a large number of Lottie users that use Carthage.

**Reply by anuragajwani** [reply]
awesome thanks, will move Lottie to Carthage then.

**Comment by jpovedap** [comment]
Hi @calda,

is this topic still a no-go?
I am currently distributing your xcframework via cocoapods (artifactory) using a forked repo.

To have this being distributed officially by airbnb would be the best case scenario. Can I help or contribute to make it possible?
I'll gladly create a PR if needed, but that will still leave you with the matter of publishing it.

**Reply by calda** [reply]
We have been distributing binary XCFrameworks in our releases for at least a year, and now publish a [lottie-spm](https://github.com/airbnb/lottie-spm) package that wraps the XCFrameworks in a way that's easy to integrate via SPM.

**Reply by jpovedap** [reply]
I know, in fact, i have been referencing https://github.com/airbnb/lottie-spm for my SPM releases.
The issue is that I haven't found a pod that exports that same XCFramework.

I download every tag's release and distribute it via Cocoapods using a podspec:

```
Pod::Spec.new do |s|
  s.name             = 'LottieXCFramework'
  s.version          = '4.4.0'
  s.summary          = 'A library to render native animations from 
bodymovin json'


  s.homepage         = 'https://github.com/airbnb/lottie-ios'
  s.license          = { :type => 'Apache', :file => 'LICENSE' }
  s.author           = { 'Brandon Withrow' => 'buba447@gmail.com', 'Cal 
Stephens' => 'cal.stephens@airbnb.com' }
  s.source           = { :git => '.' }

  s.swift_version = '5.5'
  s.ios.deployment_target = '11.0'
  s.osx.deployment_target = '10.12'
  s.tvos.deployment_target = '11.0'

  s.ios.vendored_frameworks = "Lottie.xcframework"

  s.ios.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.tvos.frameworks = ['UIKit', 'CoreGraphics', 'QuartzCore']
  s.osx.frameworks = ['AppKit', 'CoreGraphics', 'QuartzCore']
  s.module_name = 'Lottie'
end

```

So when someone who uses my library also has official's Cocoapods - Lottie as a dependency, the project doesn't compile because there is a duplicated framework.

**Reply by calda** [reply]
I would like to avoid making any changes or additions to how we distribute Lottie via Cocoapods.
